spec.md

```diff
@@ -218,9 +218,10 @@ Externally defined modules are modules whose implementation is not provided in
 the current circuit.  Only the ports and name of the externally defined module
 are specified in the circuit.  An externally defined module may include, in
 order, an optional _defname_ which sets the name of the external module in the
-resulting Verilog and zero or more name--value _parameter_ statements.  Each
-name--value parameter statement will result in a value being passed to the named
-parameter in the resulting Verilog.
+resulting Verilog, zero or more name--value _parameter_ statements, and zero or
+more _ref_ statements indicating the resolved paths of the module's exported
+references.  Each name--value parameter statement will result in a value being
+passed to the named parameter in the resulting Verilog.
```

Comment:
Good question.  I agree it can go either way, I left it optional as:
1) Only needed if the references are /used/, similar to how we allow input refs as long as they're not used for upwards references.
2) Leaves door open for not specifying and using, e.g., a linker or other design-integration step in the build flow to resolve.  This would need to happen before emitting Verilog, as you say, perhaps (for something using CIRCT) at the MLIR level.  But the point is that it could be addressed a different way, maybe.

As I write this, (2) is maybe less compelling than I thought but :shrug:.  Happy to go either way, and will clarify it regardless :+1:.

Reply:
Added language clarifying they're required if references are used.